src/common/runtime/src/global.rs

```diff
@@ -44,7 +48,8 @@ impl Default for RuntimeOptions {
         Self {
             read_rt_size: cpus,
             write_rt_size: cpus,
-            bg_rt_size: cpus,
+            bg_rt_size: usize::max(cpus / 2, 1),
+            hb_rt_size: 1,
```

Comment:
Using a dedicated runtime for heartbeat seems reasonable to me, considering that heartbeat plays a critical role in our disaster recovery. However, setting the default size to 1 is a little dangerous in that it's easy for some async tasks to block the whole runtime. It's commonly seen in tests(that's why some tests are annotated with `#[tokio::test(flavor = "multi_thread")]` instead of just `#[tokio::test]`). So I say set it to 2 at least.

On the other hand, I'm neutral to halve the bg runtime size. Adjusting runtime size like this all start from empirical guessing. What we ought to do is monitoring each runtime's "busyness". We can do that through `tokio-metric` (it's already in our codes, introduced in https://github.com/GreptimeTeam/greptimedb/pull/2985). This is a screenshot from an exported metrics from our customer about the tokio runtime metrics:
<img width="484" alt="image" src="https://github.com/GreptimeTeam/greptimedb/assets/990479/2c538052-e0f5-4fac-ac4b-2651cd34ee09">

Reply:
The reason why I set `bg_worker` to half of the cpu numbers is that we observed on production there are chances that cpu-intensive tasks, like flush/compaction, filling up all `bg_worker` runtime. And because `bg_worker` runtime has same size with cpu cores, it takes all CPU and server is failed to response heartbeat or even health check calls.

By updating the size, I want to make sure even for the worst case, we will still have CPU cores for dealing with incoming requests.

Reply:
A busy `bg_worker` runtime with the size of cpu cores does not mean the cpu cannot schedule other runtime's tasks. I think separating a dedicated runtime for heartbeat tasks from bg runtime is enough. Besides, there are global read and write runtimes, both with the size of cpu cores, could be busy as well.

Reply:
It's true. But I still believe we should limit concurrency for `bgworker` because it's shared and some of its tasks are cpu bound. For worst cases, it still gets chance to affect foreground tasks. We do witness this on our cloud deployment that `bgworker` takes all the cpus and foreground tasks like axum server failed to respond incoming requests.